examples/jsm/exporters/GLTFExporter.js

```diff
@@ -1730,6 +1788,55 @@ class GLTFWriter {
 
 	}
 
+	/**
+	 * If a vertex attribute with a
+	 * [non-standard data type](https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#meshes-overview)
+	 * is used, then it is assumed that this is an allowed data type according to the
+	 * [KHR_mesh_quantization](https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_mesh_quantization/README.md)
+	 * extension.
+	 * In this case the extension is automatically added to the list of used extensions.
+	 *
+	 * @param {string} attributeName
+	 * @param {THREE.BufferAttribute} attribute
+	 */
+	detectMeshQuantization( attributeName, attribute ) {
+
+		if ( this.extensionsUsed[ KHR_MESH_QUANTIZATION ] ) return;
+
+		let useMeshQuantization = false;
+
+		const isNonFloatAttribute = attribute.array.constructor !== Float32Array;
```

Comment:
To avoid adding KHR_mesh_quantization for assets with vertex attributes that aren't valid under _both_ the core spec and the extension, perhaps we should check that the component size is <= 2 here, instead of checking against float32? Hoping rule out uint32 and int32 types as well.

Reply:
see https://github.com/mrdoob/three.js/pull/25297#discussion_r1089028220